operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1673 @@
+		svc = svc.DeepCopy()
+
+		ipam.Logger.Debugf("Deleted service '%s/%s'", svc.GetNamespace(), svc.GetName())
+
+		ipam.handleDeletedService(ctx, svc)
+
+		// Removing a service might free up IPs which unsatisfied services are waiting for.
+		ipam.satisfyServices(ctx)
+		ipam.updateAllPoolCounts(ctx)
+
+		return nil
+	}
+}
+
+// handleUpsertService updates the service view in the service store, it removes any allocation and ingress that
+// do not belong on the service and will move the service to the satisfied or unsatisfied service view store depending
+// on if the service requests are satisfied or not.
+func (ipam *LBIPAM) handleUpsertService(ctx context.Context, svc *core_v1.Service) {
+	if svc.GetUID() == "" {
+		// The code keys everything on UIDs, without one we can't function.
+		// The APIServer should always assign one, but one tends to forget in tests
+		ipam.Logger.Error("Service has no UID!")
+		return
+	}
+
+	sv, found, _ := ipam.ServiceStore.GetService(svc.GetUID())
+	if !found {
+		sv = &ServiceView{
+			UID: svc.GetUID(),
+			Key: resource.NewKey(svc),
+		}
+	}
+
+	// Ignore services which are not meant for us
+	if !ipam.isResponsibleForSVC(svc) {
+		if !found {
+			return
+		}
+
+		// Release allocations
+		for _, alloc := range sv.AllocatedIPs {
+			alloc.Origin.allocRange.Release(alloc.IP)
+		}
+		ipam.ServiceStore.Delete(sv.UID)
+
+		// Remove all ingress IPs
+		sv.Status.LoadBalancer.Ingress = nil
+		for i := len(sv.Status.Conditions) - 1; i >= 0; i-- {
+			if sv.Status.Conditions[i].Type == ciliumSvcRequestSatisfiedCondition {
+				sv.Status.Conditions = slices.Delete(sv.Status.Conditions, i, i+1)
+			}
+		}
+
+		err := ipam.patchSvcStatus(ctx, sv)
+		if err != nil {
+			ipam.Logger.WithError(err).Error("Error while patching status (upsert clear)")
+		}
+
+		return
+	}
+
+	// We are responsible for this service.
+
+	// Update the service view
+	sv.Generation = svc.Generation
+	sv.Labels = svcLabels(svc)
+	sv.RequestedFamilies.IPv4, sv.RequestedFamilies.IPv6 = ipam.serviceIPFamilyRequest(svc)
+	sv.RequestedIPs = getSVCRequestedIPs(svc)
+	sv.Status = svc.Status.DeepCopy()
+
+	// Remove any allocation that are no longer valid due to a change in the service spec
+	ipam.stripInvalidAllocations(sv)
+
+	// Check for each ingress, if its IP has been allocated by us. If it isn't check if we can allocate that IP.
+	// If we can't, strip the ingress from the service.
+	svModifiedStatus := ipam.stripOrImportIngresses(sv)
+
+	// Attempt to satisfy this service in particular now. We do this now instread of relying on
+	// ipam.satisfyServices to avoid updating the service twice in quick succession.
+	if !sv.isSatisfied() {
+		if ipam.satisfyService(sv) {
+			svModifiedStatus = true
+		}
+	}
+
+	// If any of the steps above changed the service object, update the object.
+	if svModifiedStatus {
+		err := ipam.patchSvcStatus(ctx, sv)
+		if err != nil {
+			ipam.Logger.WithError(err).Error("Error while patching status (upsert)")
```

Comment:
Why is this and many other errors ignored? Here it seems like we should bubble the error up via the handler so it gets retried few times.